packages/lesswrong/lib/collections/tags/schema.ts

```diff
@@ -454,8 +454,68 @@ export const schema: SchemaType<DbTag> = {
     optional: true,
     ...schemaDefaultValue(false),
   },
+  parentTagId: {
+    ...foreignKeyField({
+      idFieldName: "parentTagId",
+      resolverName: "parentTag",
+      collectionName: "Tags",
+      type: "Tag",
+    }),
+    optional: true,
+    viewableBy: ['guests'],
+    editableBy: ['sunshineRegiment', 'admins'],
+    insertableBy: ['sunshineRegiment', 'admins'],
+    label: "Parent Tag",
+    tooltip: "Parent tag which will also be applied whenever this tag is applied to a post for the first time",
+    group: formGroups.advancedOptions,
+    control: 'TagSelect',
+    onCreate: async ({newDocument: tag, context }) => {
+      if (tag.parentTagId) {
+        // don't allow chained parent tag relationships
+        const { Tags } = context;
+        if ((await Tags.find({parentTagId: tag._id}).count())) {
+          throw Error(`Tag ${tag.name} is a parent tag of another tag.`);
+        }
+      }
+      return tag.parentTagId
+    },
```

Comment:
This was previously an `onInsert` method, but `onInsert` doesn't expose the `context` object, which is needed to avoid a dependency cycle. `onInsert` though is just a [deprecated version](https://blog.vulcanjs.org/vulcan-js-1-12-a-better-graphql-api-for-crud-operations-fae82e03a7e9) of `onCreate`.